docs/book/src/faq.md

```diff
@@ -0,0 +1,88 @@
+
+# Kubebuilder FAQ
+
+## How to undestand the domain field?
+First of all, we should get familiar with the GVK terminology, check [here](./cronjob-tutorial/gvks.md).
+
+The domain is for the group suffix, to explicitly show the resource group category.
+For example, if set `--domain=example.com`:
+```
+kubebuilder init --domain example.com --repo xxx --plugins=go/v4-alpha
+kubebuilder create api --group mygroup --version v1beta1 --kind Mykind
+```
+Then the result resource group will be `mygroup.example.com`.
+
+If domain field not set, the default value is `my.domain`.
+
+## How to validate API or set default value?
```

Comment:
That is not the only way to do that.
We can use the OpenAPI schema 
And to set default values we can provide CR as an example for the users as we can also set them via the controller's reconciliation. (if/when they are not provided). Have we had any issues with this one? If so, what is the issue? Could we discuss it separated so that we can move with this PR with the other things? COuld we have a follow up PR to add this one if you see that it still fit?

Reply:
Hi @astraw99,

That is not the only way to do so. I think if we would like to cover this topic we need to give further info and explain how to achieve it with all option. Also, we have a webhooks section in the tutorial guides and etc. We can add the info about how to use "--programmatic-validation" option to validate the API there as an example. 

The FAQ section would be for things like "I am facing an issue X how to solve" (when making no sense we change the code OR it is no longer faced in new versions).